hooks/use-on-click-outside.ts

```diff
@@ -1,18 +1,12 @@
 import { useEffect, useRef } from '@wordpress/element';
 
-/**
- * useOnClickOutside
- *
- * @param {Function} onClickOutside callback that will get invoked when the user clicks outside of the target
- * @returns {object} ref to the target element
- */
-export function useOnClickOutside(onClickOutside) {
-	const ref = useRef();
+export function useOnClickOutside(onClickOutside: (event: MouseEvent | TouchEvent) => void) {
```

Comment:
@fabiankaegy By typing the `event` object with `React.MouseEvent<HTMLElement> | React.TouchEvent<HTMLElement>` types you can leverage [React's synthetic event system](https://react.dev/reference/react-dom/components/common#react-event-object).